### PR TITLE
[Proposal] Kotlin extensions

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] ~Did you check UI tests on the sample app? They are not executed on CI.~

## Description

Kotlin becomes standard for Android development (see [Google I/O 2019: Android development will become increasingly Kotlin-first](https://android-developers.googleblog.com/2019/05/google-io-2019-empowering-developers-to-build-experiences-on-Android-Play.html)) and we can provide a better experience with our SDK using built-in language functions. Here is a proposal of how we can do it. It's more like a topic to discuss than ready for merge solution.

So, this PR:
* Provides Kotlin specific extensions to improve user experience.
* Uses Kotlin coroutines as a lightweight and standard way for async code. It makes the code much cleaner.
* Removes ugly `::class.java` from start call.

### Usage of SDK (before)
```kotlin
AppCenter.start(application, APP_SECRET, Analytics::class.java, Auth::class.java);
AppCenter.getInstallId().thenAccept { installId ->
    Log.d(LOG_TAG, "Install ID: $installId")
}
Auth.signIn().thenAccept { signInResult ->
    if (signInResult.exception == null) {
        val accountId = signInResult.userInformation.accountId
        Log.d(LOG_TAG, "User sign-in succeeded. Account Id: $accountId")
    } else {
        val signInFailureException = signInResult.exception
        Log.d(LOG_TAG, "User sign-in failed. Exception: $signInFailureException")
    }
}
```

### Usage of SDK (after)
```kotlin
AppCenter.start(application, APP_SECRET, Analytics, Auth)
Log.d(LOG_TAG, "Install ID: " + AppCenter.getInstallId())
try {
    val userInformation = Auth.signIn()
    Log.d(LOG_TAG, "User sign-in succeeded. Account Id: ${userInformation.accountId}")
} catch (exception: Exception) {
    Log.d(LOG_TAG, "User sign-in failed. Exception: ${exception.message}")
}
```

[AB#64380](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/64380)

### CODE REVIEW

File: apps/sasquatch-kotlin/build.gradle [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (guperrot) [inline apps/sasquatch-kotlin/build.gradle]: Ideally we just migrate the sasquatch little by little. Maintaining 2 apps is not what we want.

File: sdk/appcenter-kotlin/src/main/java/com/microsoft/appcenter/kotlin/Analytics.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    suspend fun setEnabled(enabled: Boolean) {
+        JavaAnalytics.setEnabled(enabled).await()
+    }
+
+    fun setTransmissionInterval(seconds: Int): Boolean = JavaAnalytics.setTransmissionInterval(seconds)
+
+    fun pause() = JavaAnalytics.pause()
+
+    fun resume() = JavaAnalytics.pause()
```
Comment (jwallra) [inline sdk/appcenter-kotlin/src/main/java/com/microsoft/appcenter/kotlin/Analytics.kt]: Looks like a copy-paste error :)

```suggestion
    fun resume() = JavaAnalytics.resume()
```

Comment (MatkovIvan) [reply sdk/appcenter-kotlin/src/main/java/com/microsoft/appcenter/kotlin/Analytics.kt]: :sweat_smile:

**Comment by guperrot** [comment]
Linked PR to board for later, closing for now.

**Pull request closed without merging**